src/control/connector.rs

```diff
@@ -127,6 +127,39 @@ pub enum Interface {
     Virtual,
     DSI,
     DPI,
+    Writeback,
+    SPI,
+    USB,
+}
+
+impl Interface {
+    /// Get interface name as string
+    pub fn as_str(&self) -> &str {
```

Comment:
This should be `&'static` as you're returning string literals, otherwise the lifetime of the returned `&str` is unnecessarily bound to `&self`.